### PR TITLE
Session feedback model

### PR DESCRIPTION
**Pull request by amickan**
- Adds a `Feedback` model and DRF create, retrieve and list endpoints as described in this [pitch](https://github.com/DIAGNijmegen/rse-roadmap/issues/2).
- Staff users are informed of new feedbacks via email. 
- The admin interface includes a link to automatically create a Github issue from the feedback (labeled as `bug`, containing the link to the feedback object in the admin and the `user_comment`).

### CODE REVIEW

File: app/grandchallenge/api/urls.py [hunk 1]
```diff
@@ -175,6 +175,7 @@
     basename="workstations-config",
 )
 router.register(r"workstations/sessions", SessionViewSet)
+router.register(r"workstations/feedback", FeedbackViewSet, basename="feedback")
```
Comment (jmsmkn) [inline app/grandchallenge/api/urls.py:178]: Nit: Could you put this in alphabetical order (above workstations/sessions)? The name should be `"workstations-feedback"` for consistency.

File: app/grandchallenge/workstations/tasks.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from django.contrib.auth import get_user_model
+from django.contrib.sites.models import Site
+from django.core.mail import send_mail
+
+from config import settings
+from grandchallenge.subdomains.utils import reverse
+
+
+def send_new_feedback_email_to_staff(feedback):
```
Comment (jmsmkn) [inline app/grandchallenge/workstations/tasks.py:9]: Nit: would be nicer in `emails.py`.

File: app/grandchallenge/workstations/models.py [hunk 3]
```diff
@@ -547,3 +557,35 @@ def save(self, *args, **kwargs) -> None:
...
+    screenshot = models.ImageField(
+        upload_to=feedback_screenshot_filepath,
+        storage=protected_s3_storage,
+        blank=True,
+    )
+    user_comment = models.TextField()
+    context = models.TextField(blank=True)
+
+    def save(self, *args, **kwargs) -> None:
+        created = self._state.adding
```
Comment (jmsmkn) [inline app/grandchallenge/workstations/models.py:581]: Nit: the variable should be `adding`, not `created` (got it confused with an attribute on the UUID Model)

File: app/grandchallenge/workstations/admin.py [hunk 4]
```diff
@@ -36,6 +42,50 @@ class SessionHistoryAdmin(SimpleHistoryAdmin):
...
+        "session__creator__username",
+        "user_comment",
+    ]
+
+    class Meta:
+        model = Feedback
+
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
+        return qs.select_related("session__creator")
```
Comment (jmsmkn) [inline app/grandchallenge/workstations/admin.py:65]: Nit: there is `list_select_related` for this https://docs.djangoproject.com/en/4.0/ref/contrib/admin/#django.contrib.admin.ModelAdmin.list_select_related

File: app/grandchallenge/workstations/admin.py [hunk 5]
```diff
@@ -36,6 +42,50 @@ class SessionHistoryAdmin(SimpleHistoryAdmin):
...
+            + reverse(
+                "admin:workstations_feedback_change",
+                kwargs={"object_id": obj.pk},
+            )
+            + "\n\n"
+            + obj.user_comment,
+        }
+        return format_html(
+            '<a href="{}">{}</a>',
+            "https://github.com/comic/grand-challenge.org/issues/new?"
```
Comment (jmsmkn) [inline app/grandchallenge/workstations/admin.py:82]: The repo should be https://github.com/diagnijmegen/rse-cirrus-core

**Review by jmsmkn — COMMENTED** [review]
Looks great! Some really minor things

File: app/grandchallenge/workstations/admin.py [hunk 6]
```diff
@@ -36,6 +42,50 @@ class SessionHistoryAdmin(SimpleHistoryAdmin):
     ]
 
 
+class FeedbackAdmin(ModelAdmin):
+    readonly_fields = (
+        "user_comment",
+        "session",
+        "screenshot",
```
Comment (jmsmkn) [inline app/grandchallenge/workstations/admin.py:49]: Can we actually view the screenshot or does this need to be added to `serving`?

Comment (amickan) [reply app/grandchallenge/workstations/admin.py:49]: Ah, sorry, you're right, this needs to be added to serving!

File: app/grandchallenge/serving/urls.py [hunk 7]
```diff
@@ -63,4 +64,8 @@ def to_url(self, value):
         ),
         serve_structured_challenge_submission_form,
     ),
+    path(
+        ("session-feedback/" "<uuid:feedback_pk>/" "<path:path>"),
```
Comment (jmsmkn) [inline app/grandchallenge/serving/urls.py:68]:
```suggestion
        "session-feedback/<uuid:feedback_pk>/<path:path>",
```

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**